### ISSUE TITLE
Is BUILD_DIR = executable.get_id() OK? For: configure_file('linker_script.ld', BUILD_DIR)

### ISSUE THREAD

**Issue by marc-hb** [mesonbuild/meson#5995]
This is hopefully just a (missing) documentation request. But maybe not.

The following code works4me. However I didn't find `executable.get_id()` in the documentation, I found it with... "trial and error". So is `executable.get_id()` the right way to do this?

I think what .get_id() returns is called the "private" (build) directory, correct? In that case is 
there _any_ correct way to use linker scripts?

```
ld_template = 'template.ld'
ld_script   = 'script.ld'

myexe = executable(
   'foo.S', 'bar.c',
   ...
   link_args : [ ..., '-Wl,--script=' + ld_script ],
   link_depends: ld_template,
)

objs_dir = myexe.get_id()
# /usr/bin/ld: ignoring invalid character `@' in script
# workaround for https://github.com/mesonbuild/meson/pull/291
objs_dir = '?'.join(objs_dir.split('@')) + '/'

configure_file(
  input: ld_template,
  output: ld_script,
  configuration : { 'OBJSDIR': objs_dir },
)
```

template.ld

```
SECTIONS
{
	.text : {
		@OBJSDIR@foo.S.obj(.text)
  ...
```

**Comment by dcbaker** [comment]
I think what you're doing is pretty much the only way to do this (I noticed the `get_id()` isn't documented too :/)

I've been thinking that it would be nice to have a `binutils` module for wrapping binutils, and a helper for configuring such scripts seems like a reasonable feature to add to that module.

**Comment by marc-hb** [comment]
Thanks @dcbaker, really appreciated. Let me rephrase to make sure we're on the same page:

- While not "official", using `.getid()` like this should be reasonably reliable and keep working in the near/medium future.
- While meson doesn't seem to know about linker scripts yet, it might help a bit more in the future? Any corresponding github issue(s) I should subscribe to?

**Comment by dcbaker** [comment]
I don't think there's a reason `get_id()` isn't documented, I submitted an MR to document it.
I've also opened an issue about the binutils module, you can see it referenced above.
